azkaban-web-server/src/main/java/azkaban/webapp/WebMetrics.java

```diff
@@ -42,13 +44,24 @@
 
   WebMetrics() {
     registry = MetricsManager.INSTANCE.getRegistry();
-    setupAllMetrics();
+    setupStaticMetrics();
   }
 
-  private void setupAllMetrics() {
+  private void setupStaticMetrics() {
     webGetCall = MetricsUtility.addMeter("Web-Get-Call-Meter", registry);
     webPostCall = MetricsUtility.addMeter("Web-Post-Call-Meter", registry);
-    MetricsUtility.addLongGauge("fetchLogLatency", logFetchLatency, registry);
+    MetricsUtility.addGauge("fetchLogLatency", registry, logFetchLatency::get);
+  }
+
+  public void addExecutorManagerMetrics(ExecutorManager executorManager) {
```

Comment:
The advantage of this strategy is to put all related metrics code in one place, which is convenient for code readers. Also, developers would be able to easily extend metrics component. As we discussed last week, the Mockito Testing will be able to evaluate if any function is called. What if we just test if metrics registration is called.